galite-core/src/main/kotlin/org/kopi/galite/report/MReport.kt

```diff
@@ -18,37 +18,976 @@
+   *
+   * @param    column        the index of the column
+   * @return    the name of the column
+   */
+  fun getColumnName(column: Int): String {
+    val label: String? = accessibleColumns[column]!!.label
+
+    if (label == null || label.isEmpty()) {
+      return ""
+    }
+    return if (accessibleColumns[column]!!.folded) label.substring(0, 1) else label
+  }
+
+  /**
+   * Makes the table of accessible columns from the columns variable
+   */
+  private fun createAccessibleTab() {
+    val columnCount = columns.size
+    var accessiblecolumnCount = 0
+
+    for (i in 0 until columnCount) {
+      if (columns[i]!!.options and Constants.CLO_HIDDEN == 0) {
+        accessiblecolumnCount += 1
+      }
+    }
+    accessibleColumns = arrayOfNulls(accessiblecolumnCount)
+    accessiblecolumnCount = 0
+    for (i in 0 until columnCount) {
+      if (columns[i]!!.options and Constants.CLO_HIDDEN == 0) {
+        accessibleColumns[accessiblecolumnCount++] = columns[i]
+      }
+    }
+  }
+
+  fun getDisplayLevels(column: Int): Int = displayLevels[column]
+
+  fun getReverseOrder(column: Int): Int = reverseOrder[column]
+
+  fun getDisplayOrder(column: Int): Int = displayOrder[column]
+
+  /**
+   * Returns the number of base rows.
+   */
+  fun getBaseRowCount(): Int = baseRows.size
+
+  /**
+   * Returns the number of visible rows.
+   */
+  fun getVisibleRowCount(): Int = visibleRows!!.size
+
+  // --------------------------------------------------------------------
+  // LISTENERS HANDLING
+  // --------------------------------------------------------------------
+  /**
+   * Adds a listener to the list that's notified each time a change
+   * to the data model occurs.
+   *
+   * @param l The ReportListener
+   */
+  fun addReportListener(l: ReportListener) = listenerList.add(ReportListener::class.java, l)
+
+  /**
+   * Removes a listener from the list that's notified each time a
+   * change to the data model occurs.
+   *
+   * @param l The ReportListener
+   */
+  fun removeReportListener(l: ReportListener) = listenerList.remove(ReportListener::class.java, l)
+
+  /**
+   * Notifies all listeners that the report model has changed.
+   */
+  protected fun fireContentChanged() {
+    val listeners = listenerList.listenerList
+    var i = listeners.size - 2
+
+    while (i >= 0) {
+      if (listeners[i] == ReportListener::class.java) {
+        (listeners[i + 1] as ReportListener).contentChanged()
+      }
+      i -= 2
+    }
+  }
+
+  // --------------------------------------------------------------------
+  // DATA MEMBERS
+  // --------------------------------------------------------------------
+  // Columns contains all columns defined by the user
+  // accessiblecolumns is a part of columns which contains only visible columns
+  private lateinit var columns : Array<VReportColumn?>    // array of column definitions
+  lateinit var accessibleColumns : Array<VReportColumn?>    // array of visible or hide columns
```

Comment:
accessibleColumns doesnt have a setter

please add 
lateinit var accessibleColumns : Array<VReportColumn?>    // array of visible or hide columns
   private set